internal/Get-FilteredRestoreFile.ps1

```diff
@@ -93,19 +94,19 @@ Tnen find the T-log backups needed to bridge the gap up until the RestorePoint
                 Write-Warning "$FunctionName - No Full backup found to anchor the restore" 
                 break
             }
+            $TlogStartLSN = $Fullbackup.LastLSN
             #This scans for striped full backups to build the results
             $Results += $SQLBackupdetails | where-object {$_.BackupTypeDescription -eq "Database" -and $_.FirstLSN -eq $FullBackup.FirstLSN}
             
             Write-Verbose "$FunctionName - Got a Full backup, now to find diffs if they exist"
             #Get latest Differential Backup
-            $DiffbackupsLSN = ($SQLBackupdetails | Where-Object {$_.BackupTypeDescription -eq 'Database Differential' -and $_.DatabaseBackupLSN -eq $Fullbackup.FirstLsn -and $_.BackupStartDate -lt $RestoreTime} | Sort-Object -Property BackupStartDate -descending | Select-Object -First 1).FirstLSN
+            $DiffbackupsLSN = ($SQLBackupdetails | Where-Object {$_.BackupTypeDescription -eq 'Database Differential' -and $_.DatabaseBackupLSN -eq $Fullbackup.CheckpointLSN -and $_.BackupStartDate -lt $RestoreTime} | Sort-Object -Property BackupStartDate -descending | Select-Object -First 1).FirstLSN
             #Scan for striped differential backups
-            $Diffbackups = $SqlBackupDetails | Where-Object {$_.BackupTypeDescription -eq 'Database Differential' -and $_.DatabaseBackupLSN -eq $Fullbackup.FirstLsn -and $_.FirstLSN -eq $DiffBackupsLSN}
-            $TlogStartlsn = 0
+            $Diffbackups = $SqlBackupDetails | Where-Object {$_.BackupTypeDescription -eq 'Database Differential' -and $_.DatabaseBackupLSN -eq $Fullbackup.CheckpointLSN -and $_.FirstLSN -eq $DiffBackupsLSN}
             if ($null -ne $Diffbackups){
                 Write-Verbose "$FunctionName - we have at least one diff so look for tlogs after the last one"
                 #If we have a Diff backup, we only need T-log backups post that point
-                $TlogStartLSN = ($DiffBackups | select-object -Property FirstLSN -first 1).FirstLSN
+                $TlogStartLSN = ($DiffBackups | select-object -Property LastLSN -First 1).LastLSN
```

Comment:
Better to have extra logs than needed in case the LSNs don't coincide nicely (ie; previous version).

If you've a case of this causing errors please provide the inputs/outputs